### PR TITLE
Improve performance of on chain transaction listing for big wallets

### PR DESCRIPTION
**Pull request by NicolasDorier**
This will improve performance dramatically for big wallets.
Before, anywhere we would query the wallet transactions, all the transactions from the wallet would be fetched from database, even if a subset of those were actually display or required.

Got reports for some install of the `/transactions` view and also the dashboard, which display only the 5 first transactions, taking 1 minute!

Breaking change:

* I needed to drop `TransactionStatus.Replaced` filter as the query database is already filtering those. While I could still support if if really needed (by falling back to the wasteful request if the filter is present), this make additional boilerplate, for a feature probably unused. If somebody complain, I will fallback on the old inefficient way. On top of it it wasn't tested.

### CODE REVIEW

File: BTCPayServer/Services/Wallets/BTCPayWallet.cs [hunk 1]
```diff
@@ -199,9 +202,74 @@ private async Task<UTXOChanges> GetUTXOChanges(DerivationStrategyBase strategy,
...
+                return lines;
+            }
+            // This call is more efficient for big wallets, as it doesn't need to load all transactions from the history
+            else
+            {
+                var response = new List<TransactionHistoryLine>();
+                await using var ctx = await NbxplorerConnectionFactory.OpenConnection();
+                var rows = await ctx.QueryAsync<(string tx_id, DateTimeOffset seen_at, string blk_id, long? blk_height, long balance_change, string asset_id, long confs)>(
+                    "SELECT r.tx_id, r.seen_at, t.blk_id, t.blk_height, r.balance_change, r.asset_id, COALESCE((SELECT height FROM get_tip('BTC')) - t.blk_height + 1, 0) AS confs " +
+                    "FROM get_wallets_recent(@walletId, @code, interval '100 years', @count, @skip) r " +
```
Comment (dennisreimann) [inline BTCPayServer/Services/Wallets/BTCPayWallet.cs:230]:
```suggestion
                    "FROM get_wallets_recent(@wallet_id, @code, interval '100 years', @count, @skip) r " +
```

File: BTCPayServer/Services/Wallets/BTCPayWallet.cs [hunk 2]
```diff
@@ -199,9 +202,74 @@ private async Task<UTXOChanges> GetUTXOChanges(DerivationStrategyBase strategy,
...
+                    {
+                        BalanceChange = row.asset_id is null ? Money.Satoshis(row.balance_change) : new AssetMoney(uint256.Parse(row.asset_id), row.balance_change),
+                        Height = row.blk_height,
+                        SeenAt = row.seen_at,
+                        TransactionId = uint256.Parse(row.tx_id),
+                        Confirmations = row.confs,
+                        BlockHash = row.blk_id is null ? null : uint256.Parse(row.blk_id)
+                    });
+                }
+                return response;
```
[the remark this answers is not in the document]
Comment (dennisreimann) [reply BTCPayServer/Services/Wallets/BTCPayWallet.cs:253]: The first line `var response = new List<TransactionHistoryLine>();` can be removed then.

**Comment by NicolasDorier** [comment]
wtf the tests pass

**Comment by dennisreimann** [comment]
The part that makes it fail for me is inside a condition, so I assume the test do not cover it.

**Comment by NicolasDorier** [comment]
@dennisreimann this code should be used by selenium test, as soon as you list transactions in the wallet. Checking what happened.

**Comment by NicolasDorier** [comment]
@dennisreimann damn we were never using the postgres backend in BTCPayServer!

**Pull request merged**